syntax = "proto3";

package stepper_service;

service StepperService {
rpc Run(StepperRequest) returns (StepperResponse);
}

message StepperRequest {
	Operation operation = 1;
	string motor_id = 2;
}

enum Operation {
	CREATE	= 0;
	READ	= 1;
	UPDATE	= 2;
	DELETE	= 3;
}

message StepperResponse {
	ResultCode code = 1;
	string message = 2;
}

enum ResultCode {
	OK				= 0;
	NOT_FOUND		= 1;
	UNSUPPORTED		= 2;
	BAD_REQUEST		= 3;
	EXCEPTION		= 4;
}